### PR TITLE
Add splunk HEC plugin

### PR DESCRIPTION
**Pull request by kousik93**

### CODE REVIEW

File: docker-image/v0.12/debian-cloudwatch/Dockerfile [hunk 1]
```diff
@@ -20,10 +20,11 @@ RUN buildDeps="sudo make gcc g++ libc-dev ruby-dev libffi-dev" \
     && gem install fluent-plugin-secure-forward \
     && gem install fluent-plugin-record-reformer \
     && gem install fluent-plugin-prometheus -v 0.4.0 \
+    && gem install fluent-plugin-splunk-hec \
     && gem install aws-sdk-core -v 2.10.50 \
     && gem install fluent-plugin-cloudwatch-logs -v 0.4.5 \
     && gem install fluent-plugin-detect-exceptions -v 0.0.9 \
-    && gem install fluent-plugin-kubernetes_metadata_filter -v 1.0.1 \
+    && gem install fluent-plugin-kubernetes_metadata_filter \
```
Comment (johngmyers) [inline docker-image/v0.12/debian-cloudwatch/Dockerfile:27]: Shouldn't we specify a version?

File: docker-image/v0.12/debian-cloudwatch/entrypoint.sh [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
    sed -i  '/FLUENT_ELASTICSEARCH_USER/d' /fluentd/etc/${FLUENTD_CONF}
 fi
 
-if [ -z ${FLUENT_ELASTICSEARCH_PASSWORD} ] ; then
+if [[ -z ${FLUENT_ELASTICSEARCH_PASSWORD} ]] ; then
    sed -i  '/FLUENT_ELASTICSEARCH_PASSWORD/d' /fluentd/etc/${FLUENTD_CONF}
 fi
 
-exec fluentd -c /fluentd/etc/${FLUENTD_CONF} -p /fluentd/plugins ${FLUENTD_OPT}
+exec fluentd -c /fluentd/etc/${FLUENTD_CONF} -p /fluentd/plugins ${FLUENTD_OPT}
```
Comment (johngmyers) [inline docker-image/v0.12/debian-cloudwatch/entrypoint.sh:13]: Inadvertently removed trailing newline

File: docker-image/v0.12/alpine-logentries/Dockerfile [hunk 3]
```diff
@@ -19,6 +19,7 @@ RUN set -ex \
     && gem install fluent-plugin-secure-forward \
     && gem install fluent-plugin-record-reformer \
     && gem install fluent-plugin-prometheus -v 0.4.0 \
+    && gem install fluent-plugin-splunk-hec \
```
Comment (johngmyers) [inline docker-image/v0.12/alpine-logentries/Dockerfile]: Shouldn't we specify a version for these?

Comment (kousik93) [reply docker-image/v0.12/alpine-logentries/Dockerfile]: Updated with specific version

**Review by johngmyers — APPROVED** [review]

**Pull request merged**